src/ocm/components/AddHosts/utils.ts

```diff
@@ -1,27 +1,22 @@
+import { isSNO } from '../../selectors';
 import Day2ClusterService from '../../services/Day2ClusterService';
 import { OcmClusterType } from './types';
 
 export const canAddHost = ({ cluster }: { cluster: OcmClusterType }) => {
-  let isAllowed = false;
-
   if (Day2ClusterService.getOpenshiftClusterId(cluster)) {
-    // Collision of "ready" states for AI and Telemetry
-    // Show the tab only if telemetry reports "ready", not the AI
     if (
-      (cluster.state === 'ready' &&
-        cluster.product?.id === 'OCP-AssistedInstall' &&
-        (!cluster.aiCluster || cluster.aiCluster.status === 'installed')) ||
-      // Backward compatibility for old AI clusters (when the subscription was created by
-      // the telemetry instead of by the AI-Service)
-      (cluster.product?.id !== 'OCP-AssistedInstall' && cluster.cloud_provider?.id === 'baremetal')
+      cluster.aiCluster &&
+      cluster.aiCluster.status === 'installed' &&
+      !isSNO(cluster.aiCluster)
     ) {
-      isAllowed = true;
-    }
-
-    if (cluster.aiCluster && cluster.aiCluster['high_availability_mode'] === 'None') {
-      isAllowed = false;
+      return true;
+    } else if (
+      (cluster.state === 'ready' && cluster.product?.id == 'OCP-AssistedInstaller') ||
+      cluster.cloud_provider?.id === 'baremetal'
```

Comment:
cluster.aiCluster might be defined && the entire condition holds && isSNO(cluster.aiCluster) === true -> BOOM!

Reply:
I think we should return false immediately if it is SNO, regardless of the status

Reply:
So like switch the order? First we check if `cluster.aiCluster` is defined, then if it's not SNO and the status last?  @jkilzi

Reply:
```javascript
if (cluster.aiCluster) {
      if (!isSNO(cluster.aiCluster)) {
        return cluster.aiCluster.status === "installed";
      } else {
        return false;
      }
    }
```

Reply:
LGTM, if we want to nest the conditions. :shrug:

Reply:
The logic stays the same -- all three conditions must be true for the function to be true.